### PR TITLE
Mp v2 create id set

### PR DESCRIPTION
**Pull request by Noy-Maimon**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/44349

## Description

1. Add a flag for getting the mp_v2_id_set.json (that's the filtered id set)
2. When using that flag,  eliminating the packs that are not supported from that id set. 
3. Eliminate the unsupported items from the id set: Dashboards, Widgets, Reports, LayoutsContainer, GenericDefinitions, GenericFields, GenericModules, GenericTypes.

### CODE REVIEW

File: demisto_sdk/commands/common/tools.py [hunk 1]
```diff
@@ -2166,3 +2167,24 @@ def get_current_repo() -> Tuple[str, str, str]:
...
+    Get the supporting marketplaces for the given content item, defined by the mp field in the metadata
+    Args:
+        file_path: path to content item in content repo
+
+    Returns:
+        list of names of supporting marketplaces (current options are marketplacev2 and xsoar)
+    """
+    file_path_parts = Path(file_path).parts
+
+    metadata_path = Path(*file_path_parts[0:2]) / METADATA_FILE_NAME
```
Comment (yaakovi) [inline demisto_sdk/commands/common/tools.py:2183]:
```suggestion
    metadata_path = Path(*file_path_parts[:2]) / METADATA_FILE_NAME
```

File: demisto_sdk/tests/test_files/DummyPack/pack_metadata.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (yaakovi) [inline demisto_sdk/tests/test_files/DummyPack/pack_metadata.json:1]: use an existing pack_metadata file for testing, no need to add this one

File: demisto_sdk/tests/test_files/DummyPackScriptIsXsoarOnly/pack_metadata.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    "email": "",
+    "created": "2020-04-14T00:00:00Z",
+    "categories": [
+        "Data Enrichment & Threat Intelligence"
+    ],
+    "tags": [],
+    "useCases": [],
+    "keywords": [],
+    "marketplaces": [
+        "xsoar"
```
Comment (yaakovi) [inline demisto_sdk/tests/test_files/DummyPackScriptIsXsoarOnly/pack_metadata.json:17]: should this include both MPs?

File: demisto_sdk/tests/test_files/DummyPackScriptIsXsoarOnly/pack_metadata.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{
+    "name": "DummyPackXsoarMPOnly",
```
Comment (yaakovi) [inline demisto_sdk/tests/test_files/DummyPackScriptIsXsoarOnly/pack_metadata.json:2]:
```suggestion
    "name": "DummyPackScriptIsXsoarOnly",
```

File: demisto_sdk/tests/test_files/integration-display-credentials-none/README.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
+Use the IPinfo.io API to get data about an IP address.
```
Comment (yaakovi) [inline demisto_sdk/tests/test_files/integration-display-credentials-none/README.md:1]: what is this file?

File: demisto_sdk/commands/common/tests/update_id_set_test.py [hunk 6]
```diff
@@ -435,6 +459,24 @@ def test_process_integration__exception():
         with pytest.raises(Exception):
             process_integration(test_file_path, True)
 
+    @staticmethod
+    def test__process_integration__marketplace_mismatch(mocker):
```
Comment (yaakovi) [inline demisto_sdk/commands/common/tests/update_id_set_test.py:463]:
```suggestion
    def test_process_integration__marketplace_mismatch(mocker):
```

File: demisto_sdk/commands/common/update_id_set.py [hunk 7]
```diff
@@ -857,7 +874,8 @@ def get_pack_metadata_data(file_path, print_logs: bool):
             'certification': 'certified' if json_data.get('support', '').lower() in ['xsoar', 'partner'] else '',
             "tags": json_data.get('tags', []),
             "use_cases": json_data.get('useCases', []),
-            "categories": json_data.get('categories', [])
+            "categories": json_data.get('categories', []),
+            "marketplaces": json_data.get('marketplaces', [])
```
Comment (yaakovi) [inline demisto_sdk/commands/common/update_id_set.py:878]:
```suggestion
            "marketplaces": json_data.get('marketplaces', []),
```

File: demisto_sdk/commands/common/update_id_set.py [hunk 8]
```diff
@@ -1012,20 +1030,58 @@ def get_depends_on(data_dict):
     return depends_on_list, command_to_integration
 
 
-def process_integration(file_path: str, print_logs: bool) -> list:
+def should_skip_item_by_mp(file_path: str, marketplace: str):
```
Comment (yaakovi) [inline demisto_sdk/commands/common/update_id_set.py:1033]: move the function before `build_tasks_graph`

File: demisto_sdk/commands/common/update_id_set.py [hunk 9]
```diff
@@ -1012,20 +1030,58 @@ def get_depends_on(data_dict):
...
+    Args:
+        file_path: path to content item
+        marketplace: the marketplace this current generated id set is for
+
+    Returns: True if should be skipped, else False
+
+    """
+    # first check, if we are using this id set for xsoar MP
+    if marketplace == MarketplaceVersions.XSOAR.value:
+        return False
```
Comment (yaakovi) [inline demisto_sdk/commands/common/update_id_set.py:1051]: remove, there might be packs for MPv2 only.

File: demisto_sdk/commands/common/update_id_set.py [hunk 10]
```diff
@@ -1012,20 +1030,58 @@ def get_depends_on(data_dict):
...
+
+    # second check, check field 'marketplaces' in the item's file
+    file_type = Path(file_path).suffix
+    try:
+        item_data = get_file(file_path, file_type)
+        if marketplace not in item_data.get('marketplaces', []):
```
Comment (yaakovi) [inline demisto_sdk/commands/common/update_id_set.py:1057]: if the field doesn't exist in the file, continue to the next check (the pack metadata).
if the field doesn't exist there too, consider it as XSOAR only

File: demisto_sdk/commands/common/update_id_set.py [hunk 11]
```diff
@@ -1012,20 +1030,58 @@ def get_depends_on(data_dict):
...
+
+    """
+    # first check, if we are using this id set for xsoar MP
+    if marketplace == MarketplaceVersions.XSOAR.value:
+        return False
+
+    # second check, check field 'marketplaces' in the item's file
+    file_type = Path(file_path).suffix
+    try:
+        item_data = get_file(file_path, file_type)
```
Comment (yaakovi) [inline demisto_sdk/commands/common/update_id_set.py:1056]: avoid opening the file twice (both here and in the `get_*_data` functions)

File: demisto_sdk/commands/common/update_id_set.py [hunk 12]
```diff
@@ -1630,7 +1732,7 @@ def re_create_id_set(id_set_path: Optional[str] = DEFAULT_ID_SET_PATH, pack_to_c
         if 'Packs' in objects_to_create:
             print_color("\nStarting iteration over Packs", LOG_COLORS.GREEN)
             for pack_data in pool.map(partial(get_pack_metadata_data,
-                                              print_logs=print_logs
+                                              print_logs=print_logs, marketplace=marketplace
```
Comment (yaakovi) [inline demisto_sdk/commands/common/update_id_set.py:1735]:
```suggestion
                                              print_logs=print_logs,
                                              marketplace=marketplace,
```

File: demisto_sdk/commands/common/update_id_set.py [hunk 13]
```diff
@@ -1640,7 +1742,7 @@ def re_create_id_set(id_set_path: Optional[str] = DEFAULT_ID_SET_PATH, pack_to_c
         if 'Integrations' in objects_to_create:
             print_color("\nStarting iteration over Integrations", LOG_COLORS.GREEN)
             for arr in pool.map(partial(process_integration,
-                                        print_logs=print_logs
+                                        print_logs=print_logs, marketplace=marketplace
```
Comment (yaakovi) [inline demisto_sdk/commands/common/update_id_set.py:1745]:
```suggestion
                                        print_logs=print_logs,
                                        marketplace=marketplace,
```

File: demisto_sdk/commands/common/update_id_set.py [hunk 14]
```diff
@@ -93,6 +94,37 @@
...
+    """
+
+    # first check, check field 'marketplaces' in the item's file
+    file_type = Path(file_path).suffix
+    try:
+        item_data = get_file(file_path, file_type)
+        if item_data.get('marketplaces'):
+            if marketplace not in item_data.get('marketplaces'):
+                return True
+    except (ValueError, FileNotFoundError, IsADirectoryError):
```
Comment (yaakovi) [inline demisto_sdk/commands/common/update_id_set.py:121]: when will you get each exception?

File: demisto_sdk/commands/create_id_set/create_id_set.py [hunk 15]
```diff
@@ -33,6 +32,10 @@ def __init__(self, output: Optional[str] = '', input: Optional[str] = None, prin
         self.id_set = OrderedDict()  # type: ignore
         self.marketplace = marketplace.lower()
 
+        # if not self.output:
+        #     self.output = MP_V2_ID_SET_PATH if self.marketplace == MarketplaceVersions.MarketplaceV2.value \
+        #         else DEFAULT_ID_SET_PATH
+
```
Comment (yaakovi) [inline demisto_sdk/commands/create_id_set/create_id_set.py:38]: remove

**Review by yaakovi — COMMENTED** [review]
nice work, what about this comment: https://github.com/demisto/demisto-sdk/pull/1745#discussion_r774124023

File: demisto_sdk/commands/common/tools.py [hunk 16]
```diff
@@ -2169,23 +2169,48 @@ def get_current_repo() -> Tuple[str, str, str]:
...
-        return marketplaces
+            marketplaces = metadata.get(MARKETPLACE_KEY_PACK_METADATA)
+            if not marketplaces:
+                return [MarketplaceVersions.XSOAR.value]
+            return marketplaces
     except FileNotFoundError:
         return []
+
+
+def get_pack_dir(path):
```
[the remark this answers is not in the document]
Comment (Noy-Maimon) [reply demisto_sdk/commands/common/tools.py:2202]: create_id_set_test has creation of temp pack, that is longer than the common path, and not always having a metadata file

File: demisto_sdk/commands/common/tools.py [hunk 17]
```diff
@@ -2169,23 +2169,48 @@ def get_current_repo() -> Tuple[str, str, str]:
...
     Args:
         file_path: path to content item in content repo
 
     Returns:
         list of names of supporting marketplaces (current options are marketplacev2 and xsoar)
     """
-    file_path_parts = Path(file_path).parts
-
-    metadata_path = Path(*file_path_parts[0:2]) / METADATA_FILE_NAME
+    if METADATA_FILE_NAME in Path(file_path).parts:  # for unified integrations
```
[the remark this answers is not in the document]
Comment (Noy-Maimon) [reply demisto_sdk/commands/common/tools.py:2182]: This comment shouldnt be there, removed 
(At first i was 'walking' backwards on the trail of the path, then changes it to forward)

File: demisto_sdk/commands/common/update_id_set.py (hunk 14 above)
Comment (Noy-Maimon) [reply demisto_sdk/commands/common/update_id_set.py:121]: First one was for files not having a valid structure, that error is being raised by get_file
second and third i got from random files in content (such as the files under content/TestPlaybooks)
we discussed that folder and we said it should be ignored

File: demisto_sdk/commands/common/tools.py [hunk 18]
```diff
@@ -2179,15 +2179,12 @@ def get_mp_types_from_metadata_by_item(file_path):
     Returns:
         list of names of supporting marketplaces (current options are marketplacev2 and xsoar)
     """
-    if METADATA_FILE_NAME in Path(file_path).parts:  # for unified integrations
+    if METADATA_FILE_NAME in Path(file_path).parts:  # for when the type is pack, the item we get is the metasata path
```
Comment (yaakovi) [inline demisto_sdk/commands/common/tools.py:2182]:
```suggestion
    if METADATA_FILE_NAME in Path(file_path).parts:  # for when the type is pack, the item we get is the metadata path
```

**Review by yaakovi — APPROVED** [review]

**Pull request merged**